worker/src/RTC/RtpPacket.cpp

```diff
@@ -600,12 +598,94 @@ namespace RTC
 
 			auto currentLen = extension->len + 1;
 
-			// Fill with 0's if new length is minor.
-			if (len < currentLen)
-				std::memset(extension->value + len, 0, currentLen - len);
+			MS_DEBUG_DEV(
+			  "set extension id: %" PRIu8 ", length:%" PRIu8 ", currentLen:%" PRIu8 " value:%s",
+			  id,
+			  len,
+			  currentLen,
+			  value.c_str());
+			
+			if (len != currentLen)
+			{
+				// If len != currentLen, means we may need to shift extension or payload.
+				uint8_t* extensionStart    = reinterpret_cast<uint8_t*>(this->headerExtension) + 4;
+				uint8_t* extensionEnd      = extensionStart + GetHeaderExtensionLength();
```

Comment:
```suggestion
				uint8_t* extensionsEnd      = extensionStart + GetHeaderExtensionLength();
```

Reply:
this line was copied from RtpPacket::ParseExtensions(), do you mean I have to change this line or all of other places?

Reply:
Ok, we can leave for for consistency, and rename it later if needed.